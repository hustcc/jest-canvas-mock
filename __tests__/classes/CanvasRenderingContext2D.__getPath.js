let ctx;
beforeEach(() => {
  // get a new context each test
  ctx = document.createElement('canvas').getContext('2d');
});

const path = new Path2D();
path.arc(100, 101, 10, 0, Math.PI * 2);

afterEach(() => {
  const drawCalls = ctx.__getPath();
  expect(drawCalls).toMatchSnapshot();
});

describe('__getPath', () => {
  it('should have a path item when arc is called', () => {
    ctx.arc(1, 2, 3, 4, 5, true);
  });

  it('should not have a path item when arc is called with a non-finite number', () => {
    ctx.arc(NaN, 2, 3, 4, 5, false);
  });

  it('should have a path item when arcTo is called', () => {
    ctx.arcTo(1, 2, 3, 4, 5);
  });

  it('should not have a path item when arcTo is called with a non-finite number', () => {
    ctx.arcTo(NaN, 2, 3, 4, 5);
  });

  it('should reset the path with beginPath', () => {
    ctx.arc(1, 2, 3, 4, 5, true);
    ctx.arc(1, 2, 3, 4, 5, true);
    ctx.arc(1, 2, 3, 4, 5, true);
    ctx.arc(1, 2, 3, 4, 5, true);
    ctx.beginPath();
  });

  it('should have a path item when bezierCurveTo is called', () => {
    ctx.bezierCurveTo(1, 2, 3, 4, 5, 6);
  });

  it('should not have a path item when bezierCurveTo is called with a non-finite number', () => {
    ctx.bezierCurveTo(NaN, 2, 3, 4, 5, 6);
  });

  it('should have a path item when clip is called', () => {
    ctx.rect(1, 2, 3, 4);
    ctx.clip();
  });

  it('should have a path item when clip is called with a fillRule', () => {
    ctx.rect(1, 2, 3, 4);
    ctx.clip('evenodd');
  });

  it('should have a path item when clip is called with a path', () => {
    ctx.clip(path);
  });

  it('should have a path item when clip is called with a path', () => {
    ctx.clip(path, 'evenodd');
  });

  it('should have a path item when closePath is called', () => {
    ctx.closePath();
  });

  it('should have a path item when ellipse is called', () => {
    ctx.ellipse(1, 2, 3, 4, 5, 6, 7, true);
  });

  it('should not have a path item when ellipse is called with non-finite numbers', () => {
    ctx.ellipse(NaN, 2, 3, 4, 5, 6, 7, false);
  });

  it('should have a path item when lineTo is called', () => {
    ctx.lineTo(1, 2);
  });

  it('should not have a path item when lineTo is called with non-finite values', () => {
    ctx.lineTo(NaN, 0);
  });

  it('should have a path item when moveTo is called', () => {
    ctx.moveTo(1, 2);
  });

  it('should not have a path item when moveTo is called with non-finite values', () => {
    ctx.moveTo(NaN, 1);
  });

  it('should have a path item when quadraticCurveTo is called', () => {
    ctx.quadraticCurveTo(1, 2, 3, 4);
  });

  it('should not have a path item when quadraticCurveTo is called with non-finite values', () => {
    ctx.quadraticCurveTo(NaN, 2, 3, 4);
  });

  it('should have a path item when rect is called', () => {
    ctx.rect(1, 2, 3, 4);
  });

  it('should not have a path item when rect is called with non-finite values', () => {
    ctx.rect(NaN, 2, 3, 4);
  });
});
