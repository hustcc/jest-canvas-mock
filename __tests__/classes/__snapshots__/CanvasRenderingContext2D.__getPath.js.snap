// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__getPath should have a path item when arc is called 1`] = `
Array [
  Object {
    "props": Object {},
    "type": "beginPath",
  },
  Object {
    "props": Object {
      "anticlockwise": true,
      "endAngle": 5,
      "radius": 3,
      "startAngle": 4,
      "x": 1,
      "y": 2,
    },
    "type": "arc",
  },
]
`;

exports[`__getPath should have a path item when arcTo is called 1`] = `
Array [
  Object {
    "props": Object {},
    "type": "beginPath",
  },
  Object {
    "props": Object {
      "cpx1": 1,
      "cpx2": 3,
      "cpy1": 2,
      "cpy2": 4,
      "radius": 5,
    },
    "type": "arcTo",
  },
]
`;

exports[`__getPath should have a path item when bezierCurveTo is called 1`] = `
Array [
  Object {
    "props": Object {},
    "type": "beginPath",
  },
  Object {
    "props": Object {
      "cpx1": 1,
      "cpx2": 3,
      "cpy1": 2,
      "cpy2": 4,
      "x": 5,
      "y": 6,
    },
    "type": "bezierCurveTo",
  },
]
`;

exports[`__getPath should have a path item when clip is called 1`] = `
Array [
  Object {
    "props": Object {},
    "type": "beginPath",
  },
  Object {
    "props": Object {
      "height": 4,
      "width": 3,
      "x": 1,
      "y": 2,
    },
    "type": "rect",
  },
  Object {
    "props": Object {
      "fillRule": "nonzero",
      "path": Array [
        Object {
          "props": Object {},
          "type": "beginPath",
        },
        Object {
          "props": Object {
            "height": 4,
            "width": 3,
            "x": 1,
            "y": 2,
          },
          "type": "rect",
        },
      ],
    },
    "type": "clip",
  },
]
`;

exports[`__getPath should have a path item when clip is called with a fillRule 1`] = `
Array [
  Object {
    "props": Object {},
    "type": "beginPath",
  },
  Object {
    "props": Object {
      "height": 4,
      "width": 3,
      "x": 1,
      "y": 2,
    },
    "type": "rect",
  },
  Object {
    "props": Object {
      "fillRule": "evenodd",
      "path": Array [
        Object {
          "props": Object {},
          "type": "beginPath",
        },
        Object {
          "props": Object {
            "height": 4,
            "width": 3,
            "x": 1,
            "y": 2,
          },
          "type": "rect",
        },
      ],
    },
    "type": "clip",
  },
]
`;

exports[`__getPath should have a path item when clip is called with a path 1`] = `
Array [
  Object {
    "props": Object {},
    "type": "beginPath",
  },
  Object {
    "props": Object {
      "fillRule": "nonzero",
      "path": Array [
        Object {
          "props": Object {
            "anticlockwise": false,
            "endAngle": 6.283185307179586,
            "radius": 10,
            "startAngle": 0,
            "x": 100,
            "y": 101,
          },
          "type": "arc",
        },
      ],
    },
    "type": "clip",
  },
]
`;

exports[`__getPath should have a path item when clip is called with a path 2`] = `
Array [
  Object {
    "props": Object {},
    "type": "beginPath",
  },
  Object {
    "props": Object {
      "fillRule": "evenodd",
      "path": Array [
        Object {
          "props": Object {
            "anticlockwise": false,
            "endAngle": 6.283185307179586,
            "radius": 10,
            "startAngle": 0,
            "x": 100,
            "y": 101,
          },
          "type": "arc",
        },
      ],
    },
    "type": "clip",
  },
]
`;

exports[`__getPath should have a path item when closePath is called 1`] = `
Array [
  Object {
    "props": Object {},
    "type": "beginPath",
  },
  Object {
    "props": Object {},
    "type": "closePath",
  },
]
`;

exports[`__getPath should have a path item when ellipse is called 1`] = `
Array [
  Object {
    "props": Object {},
    "type": "beginPath",
  },
  Object {
    "props": Object {
      "anticlockwise": true,
      "endAngle": 7,
      "radiusX": 3,
      "radiusY": 4,
      "rotation": 5,
      "startAngle": 6,
      "x": 1,
      "y": 2,
    },
    "type": "ellipse",
  },
]
`;

exports[`__getPath should have a path item when lineTo is called 1`] = `
Array [
  Object {
    "props": Object {},
    "type": "beginPath",
  },
  Object {
    "props": Object {
      "x": 1,
      "y": 2,
    },
    "type": "lineTo",
  },
]
`;

exports[`__getPath should have a path item when moveTo is called 1`] = `
Array [
  Object {
    "props": Object {},
    "type": "beginPath",
  },
  Object {
    "props": Object {
      "x": 1,
      "y": 2,
    },
    "type": "moveTo",
  },
]
`;

exports[`__getPath should have a path item when quadraticCurveTo is called 1`] = `
Array [
  Object {
    "props": Object {},
    "type": "beginPath",
  },
]
`;

exports[`__getPath should have a path item when rect is called 1`] = `
Array [
  Object {
    "props": Object {},
    "type": "beginPath",
  },
  Object {
    "props": Object {
      "height": 4,
      "width": 3,
      "x": 1,
      "y": 2,
    },
    "type": "rect",
  },
]
`;

exports[`__getPath should not have a path item when arc is called with a non-finite number 1`] = `
Array [
  Object {
    "props": Object {},
    "type": "beginPath",
  },
]
`;

exports[`__getPath should not have a path item when arcTo is called with a non-finite number 1`] = `
Array [
  Object {
    "props": Object {},
    "type": "beginPath",
  },
]
`;

exports[`__getPath should not have a path item when bezierCurveTo is called with a non-finite number 1`] = `
Array [
  Object {
    "props": Object {},
    "type": "beginPath",
  },
]
`;

exports[`__getPath should not have a path item when ellipse is called with non-finite numbers 1`] = `
Array [
  Object {
    "props": Object {},
    "type": "beginPath",
  },
]
`;

exports[`__getPath should not have a path item when lineTo is called with non-finite values 1`] = `
Array [
  Object {
    "props": Object {},
    "type": "beginPath",
  },
]
`;

exports[`__getPath should not have a path item when moveTo is called with non-finite values 1`] = `
Array [
  Object {
    "props": Object {},
    "type": "beginPath",
  },
]
`;

exports[`__getPath should not have a path item when quadraticCurveTo is called with non-finite values 1`] = `
Array [
  Object {
    "props": Object {},
    "type": "beginPath",
  },
]
`;

exports[`__getPath should not have a path item when rect is called with non-finite values 1`] = `
Array [
  Object {
    "props": Object {},
    "type": "beginPath",
  },
]
`;

exports[`__getPath should reset the path with beginPath 1`] = `
Array [
  Object {
    "props": Object {},
    "type": "beginPath",
  },
]
`;
